base/compiler/abstractinterpretation.jl

```diff
@@ -1055,10 +1081,14 @@ function abstract_call(interp::AbstractInterpreter, fargs::Union{Nothing,Vector{
         f = ft.parameters[1]
     elseif isa(ft, DataType) && isdefined(ft, :instance)
         f = ft.instance
+    elseif isa(ft, PartialOpaque)
+        return abstract_call_opaque_closure(interp, ft, argtypes, sv)
+    elseif isa(ft, DataType) && unwrap_unionall(ft).name === typename(Core.OpaqueClosure)
+        return CallMeta(rewrap_unionall(unwrap_unionall(ft).parameters[2], ft), false)
     else
         # non-constant function, but the number of arguments is known
         # and the ft is not a Builtin or IntrinsicFunction
-        if typeintersect(widenconst(ft), Builtin) != Union{}
+        if typeintersect(widenconst(ft), Union{Builtin, Core.OpaqueClosure}) != Union{}
```

Comment:
Seems like we should consider making it a Builtin?

Reply:
Potentially? It's kinda weird to call it a builtin though. Maybe add something above builtin?

Reply:
I think it is perhaps accurate enough calling it a Builtin—there is no source code since that is in C and uses a JL_CALLABLE handle for the initial `invoke`, and there's otherwise about a dozen spots in compiler similar to this one that we need to audit. OTOH, in C we assume that Builtin must have exactly one entry in the method table cache (which will be false for the OpaqueClosure), so perhaps we need to audit each anyways.

Reply:
Do we still need to audit these, or did you already look through the other places where we treat it specially?

Reply:
I left this as is for the moment. It still seems wrong to be to call it a `Builtin`. We could perhaps insert a new abstract type above `Builtin` though.

Reply:
I don't think that is necessarily useful. We might want to make an alias name for this Union in Core.Compiler, but currently this is the only usage site. I was inquiring instead whether any of the other places that we refer to Builtin though need to use this Union instead.

Reply:
I looked through those and I didn't think any needed to be adjusted.

Reply:
Ah, good. Thanks for confirming